cmd/pbstatuser/main.go

```diff
@@ -131,9 +146,11 @@ func checkSourceAvailabilityAWS(ctx context.Context) {
 			sr.Err = err
 			logger.Warn().Msgf("Could not get aws assumed client %s", err)
 			chSend <- sr
+			totalAvailabilityCheckReqs.WithLabelValues("aws", string(sr.Status), err.Error()).Inc()
 		} else {
 			sr.Status = kafka.StatusAvaliable
 			chSend <- sr
+			totalAvailabilityCheckReqs.WithLabelValues("aws", string(sr.Status), "").Inc()
```

Comment:
Can't you call this just once passing `err` to the method? Hopefully the Prom client will not panic if error is `nil`?

Reply:
When the error is nil I am getting a null pointer exception, I'll see if there is maybe a different usage that can fit.

Reply:
It makes sense why we need to differentiate between the cases. 
first of all the label parameters are of type string (error type is not possible). 
second,  when
 ```golang
 err != nil
```
we have an error thus a  string error when calling err.Error,  but when we do not have an error meaning 
```golang
 err = nil
``` 
We are calling nil.Error() which will obviously causes a null pointer exception.

Reply:
What I mean was to use `error` type for the argument, therefore you do not need to call `err.Error()` at all.